share/spack/docker/package-index/Dockerfile

```diff
@@ -0,0 +1,31 @@
+# prepare the package index in form of JSON files
+FROM ubuntu:18.04 AS build-env
+
+ENV SPACK_ROOT=/opt/spack             \
+    DEBIAN_FRONTEND=noninteractive
+
+COPY bin     $SPACK_ROOT/bin
+COPY etc     $SPACK_ROOT/etc
+COPY lib     $SPACK_ROOT/lib
+COPY share   $SPACK_ROOT/share
+COPY var     $SPACK_ROOT/var
+
+RUN apt-get -yqq update   \
+ && apt-get -yqq install  \
+        bash jq python    \
+ && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /build
+# single, large index file
+RUN $SPACK_ROOT/bin/spack list --format version_json > packages.json
+# individual packages split into a tree of :firstLetter/:packageName.json
+COPY share/spack/docker/package-index/split.sh ./
+RUN ./split.sh
```

Comment:
```diff
--- COPY share/spack/docker/package-index/split.sh ./
--- RUN ./split.sh
+++ RUN $SPACK_ROOT/share/spack/docker/package-index/split.sh
```

Reply:
Really, that works? It's an external script o.0

Reply:
ah no, I copy all things in xD